File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/bootyisland/domain/BootyIslandObject.java
package com.zwstudio.logicpuzzlesandroid.puzzles.bootyisland.domain;


public abstract class BootyIslandObject {
    public abstract String objAsString();
    public static BootyIslandObject objFromString(String str) {
        switch (str) {
            case "marker":
                return new BootyIslandMarkerObject();
            case "treasure":
                return new BootyIslandTreasureObject();
            default:
                return new BootyIslandMarkerObject();
        }
    }
}

File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/bootyisland/BootyIslandObject.kt
package com.zwstudio.logicpuzzlesandroid.puzzles.bootyisland

import com.zwstudio.logicpuzzlesandroid.common.domain.Position


enum class BootyIslandObject {
    Empty, Forbidden, Hint, Marker, Treasure
}

class BootyIslandGameMove(val p: Position, var obj: BootyIslandObject = BootyIslandObject.Empty)
